src/Metadata/Extractor/YamlResourceExtractor.php

```diff
@@ -0,0 +1,351 @@
+            'description' => $this->phpize($resource, 'description', 'string'),
+            'urlGenerationStrategy' => $this->phpize($resource, 'urlGenerationStrategy', 'integer'),
+            'deprecationReason' => $this->phpize($resource, 'deprecationReason', 'string'),
+            'elasticsearch' => $this->phpize($resource, 'elasticsearch', 'bool'),
+            'fetchPartial' => $this->phpize($resource, 'fetchPartial', 'bool'),
+            'forceEager' => $this->phpize($resource, 'forceEager', 'bool'),
+            'paginationClientEnabled' => $this->phpize($resource, 'paginationClientEnabled', 'bool'),
+            'paginationClientItemsPerPage' => $this->phpize($resource, 'paginationClientItemsPerPage', 'bool'),
+            'paginationClientPartial' => $this->phpize($resource, 'paginationClientPartial', 'bool'),
+            'paginationEnabled' => $this->phpize($resource, 'paginationEnabled', 'bool'),
+            'paginationFetchJoinCollection' => $this->phpize($resource, 'paginationFetchJoinCollection', 'bool'),
+            'paginationUseOutputWalkers' => $this->phpize($resource, 'paginationUseOutputWalkers', 'bool'),
+            'paginationItemsPerPage' => $this->phpize($resource, 'paginationItemsPerPage', 'integer'),
+            'paginationMaximumItemsPerPage' => $this->phpize($resource, 'paginationMaximumItemsPerPage', 'integer'),
+            'paginationPartial' => $this->phpize($resource, 'paginationPartial', 'bool'),
+            'paginationType' => $this->phpize($resource, 'paginationType', 'string'),
+            'security' => $this->phpize($resource, 'security', 'string'),
+            'securityMessage' => $this->phpize($resource, 'securityMessage', 'string'),
+            'securityPostDenormalize' => $this->phpize($resource, 'securityPostDenormalize', 'string'),
+            'securityPostDenormalizeMessage' => $this->phpize($resource, 'securityPostDenormalizeMessage', 'string'),
+            'securityPostValidation' => $this->phpize($resource, 'securityPostValidation', 'string'),
+            'securityPostValidationMessage' => $this->phpize($resource, 'securityPostValidationMessage', 'string'),
+            'input' => $this->phpize($resource, 'input', 'string'),
+            'output' => $this->phpize($resource, 'output', 'string'),
+            'normalizationContext' => $this->buildAttribute($resource, 'normalizationContext'),
+            'denormalizationContext' => $this->buildAttribute($resource, 'denormalizationContext'),
+            'validationContext' => $this->buildAttribute($resource, 'validationContext'),
+            'filters' => $this->buildAttribute($resource, 'filters'),
+            'order' => $this->buildAttribute($resource, 'order'),
+            'extraProperties' => $this->buildAttribute($resource, 'extraProperties'),
+            'mercure' => $this->buildMercure($resource),
+            'messenger' => $this->buildMessenger($resource),
+        ];
+    }
```

Comment:
Do you think that we could use the `Metadata/ApiResource` class to build these? I'm thinking of something using reflection:

```
      $refl = new \ReflectionClass(ApiResource::class);
      foreach($refl->getMethods() as $method) {
        if(0 !== strpos($method->getName(), 'get')) {continue;}
        $this->phpize($resource, strtolower(str_replace('get', $method->getName())), $method->getReturnType()); //also replace `?` in the return type
      }
```

we'd probably need to check filters, mercure and messenger. Then `buildAttributes` could be inside `phpize` when the type is an array.

Reply:
I will still need to filter which property is in the `buildBase` method or not. Those properties can be declared at the resource level, but also overriden by operation, and not all properties can be declared on both sides. That's why I had to filter them here